playbooks/roles/ocp-config/tasks/main.yaml

```diff
@@ -56,6 +56,14 @@
     args:
       chdir: "{{ workdir }}"
 
+  - name: Generate Machine Config Master and Worker nodes
+    when:
+      - luks.enabled and luks is defined
+      - luks.config | length > 0
+      - luks.config[0].thumbprint is not none
+      - luks.config[0].url is not none
```

Comment:
Does not make sense just to check the 0th element here. These can be removed as you have checks in the role.

Reply:
We have added conditions
```
      - luks.config[0].thumbprint is not none
      - luks.config[0].url is not none
```
to check if user has provided both thumbprint as well as url atleast for index 0 so that deployment will not fail and for index 1 and so we have added condition in template file [playbooks/roles/ocp-config/templates/99-node-machineconfig.yaml.j2](https://github.com/ocp-power-automation/ocp4-playbooks/pull/173/files#diff-cf1867fb3b61afc6dfefa111a398e385dbf46064b4012d509ec9942e05c13eac) as follows
```
{% for item in luks.config  %}
{% if item.thumbprint is not none and item.url is not none %}
              - thumbprint: {{ item.thumbprint}}
                url: {{ item.url }}
{% endif %}
```
Because if user failed to add appropriate values at index 0 the cluster deployment will get failed and it will be in stuck state as no node will get rebooted.